### PR TITLE
 Skeleton DNS feature and StratisDnsD with test project

### PR DESCRIPTION
**Pull request by 345James**
Created a StratisDnsD daemon project that by default runs the base feature, consensus (for RPC), API, RPC and DNS features.

- New DNS feature project and corresponding test project (with minimal unit tests to cover what is there).
- Added a reference to the DNS nuget package which will be used by the DNS feature in future commits.

### CODE REVIEW

**Comment by 345James** [comment]
[[DNS] Skeleton DNS feature and StratisDnsD with test project](https://trello.com/c/4EOMHOf3/326-dns-skeleton-dns-feature-and-stratisdnsd-with-test-project)

File: src/Stratis.StratisDnsD/Program.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        static void Main(string[] args)
+        {
+            MainAsync(args).Wait();
+        }
+
+        public static async Task MainAsync(string[] args)
+        {
+            try
+            {
+                Network network = args.Contains("-testnet") ? Network.StratisTest : Network.StratisMain;
```
Comment (bokobza) [inline src/Stratis.StratisDnsD/Program.cs]: The network shouldn't be limited to Stratis.
Have a look in BreezeD, there we support stratis or bitcoin: https://github.com/stratisproject/StratisBitcoinFullNode/blob/master/src/Stratis.BreezeD/Program.cs#L34.
So for stratis you'd do `dotnet run stratis -testnet` and bitcoin you'd do `dotnet run -testnet`.

Comment (bokobza) [reply src/Stratis.StratisDnsD/Program.cs]: Just realized that this project is called Stratis.StratisDnsD so maybe your intention was indeed to create a stratis only project.
How about making it network agnostic and calling it Stratis.DnsD?

File: src/Stratis.Bitcoin.Features.Dns.Tests/GivenADnsFeature.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+namespace Stratis.Bitcoin.Features.Dns.Tests
+{
+    /// <summary>
+    /// Tests for the <see cref="DnsFeature"/> class.
+    /// </summary>
+    public class GivenADnsFeature
+    {
+        [Fact]
+        [Trait("UnitTest", "UnitTest")]
```
Comment (bokobza) [inline src/Stratis.Bitcoin.Features.Dns.Tests/GivenADnsFeature.cs]: Would naming the trait `[Trait("Dns", "UnitTest")]` be more useful?

Comment (345James) [reply src/Stratis.Bitcoin.Features.Dns.Tests/GivenADnsFeature.cs]: Good point, makes it easier then to just run the Dns tests.

File: src/Stratis.StratisDnsD/Program.cs (hunk 1 above)
Comment (345James) [reply src/Stratis.StratisDnsD/Program.cs]: I have just checked and I believe the decision on Monday was to limit this to only the Stratis network. The PR was created with this in mind.

Comment (bokobza) [reply src/Stratis.StratisDnsD/Program.cs]: Ok cool mate, that's fair enough.

**Pull request merged**